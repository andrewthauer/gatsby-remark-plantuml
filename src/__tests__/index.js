jest.mock(`gatsby-cli/lib/reporter`, () => {
  return {
    panicOnBuild: jest.fn(),
    panic: jest.fn(),
    error: jest.fn(),
    warn: jest.fn(),
    info: jest.fn(),
  }
})

const remark = require(`remark`)
const reporter = require(`gatsby-cli/lib/reporter`)

let plugin

beforeEach(() => {
  Object.keys(reporter).forEach(key => reporter[key].mockReset())
})

describe(`remark plantuml plugin`, () => {
  beforeEach(() => {
    jest.resetModules()
    plugin = require(`../index`)
  })

  describe(`ignores codeblocks that are not plantuml`, () => {
    it(`with lang=''`, () => {
      const code = `\`\`\`\n// Fake\n\`\`\``
      const markdownAST = remark.parse(code)
      return plugin({ markdownAST, reporter }).then(() => {
        expect(markdownAST).toMatchSnapshot()
        expect(reporter.info).not.toHaveBeenCalled()
        expect(reporter.warn).not.toHaveBeenCalled()
        expect(reporter.error).not.toHaveBeenCalled()
        expect(reporter.panic).not.toHaveBeenCalled()
        expect(reporter.panicOnBuild).not.toHaveBeenCalled()
      })
    })

    it(`with lang='js'`, () => {
      const code = `\`\`\`js\n// Fake\n\`\`\``
      const markdownAST = remark.parse(code)
      return plugin({ markdownAST, reporter }).then(() => {
        expect(markdownAST).toMatchSnapshot()
        expect(reporter.info).not.toHaveBeenCalled()
        expect(reporter.warn).not.toHaveBeenCalled()
        expect(reporter.error).not.toHaveBeenCalled()
        expect(reporter.panic).not.toHaveBeenCalled()
        expect(reporter.panicOnBuild).not.toHaveBeenCalled()
      })
    })
  })

  it(`fails the build when can't find Java and Graphviz`, () => {
    const path = process.env.PATH
    process.env.PATH = `` // an empty PATH won't find any files
    const code = `
\`\`\`plantuml
@startuml
Alice -> Bob: Authentication Request
Bob --> Alice: Authentication Response

Alice -> Bob: Another authentication Request
Alice <-- Bob: Another authentication Response
@enduml
\`\`\`
`
    const markdownAST = remark.parse(code)
    return plugin({ markdownAST, reporter })
      .then(() => {
        expect(markdownAST).toMatchSnapshot()
        expect(reporter.info).not.toHaveBeenCalled()
        expect(reporter.warn).toHaveBeenCalledTimes(2)
        expect(reporter.warn).toHaveBeenNthCalledWith(
          1,
          `Executable 'java' not found on path`
        )
        expect(reporter.warn).toHaveBeenNthCalledWith(
          2,
          `Executable 'dot' not found on path`
        )
        expect(reporter.error).not.toHaveBeenCalled()
        expect(reporter.panic).not.toHaveBeenCalled()
        expect(reporter.panicOnBuild).not.toHaveBeenCalled()
      })
      .finally(() => {
        process.env.PATH = path
      })
  })
})
