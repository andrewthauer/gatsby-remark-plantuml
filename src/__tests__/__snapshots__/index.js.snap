// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`remark plantuml plugin fails the build when can't find Java and Graphviz 1`] = `
Object {
  "children": Array [
    Object {
      "lang": "plantuml",
      "position": Position {
        "end": Object {
          "column": 4,
          "line": 10,
          "offset": 203,
        },
        "indent": Array [
          1,
          1,
          1,
          1,
          1,
          1,
          1,
          1,
        ],
        "start": Object {
          "column": 1,
          "line": 2,
          "offset": 1,
        },
      },
      "type": "code",
      "value": "@startuml
Alice -> Bob: Authentication Request
Bob --> Alice: Authentication Response

Alice -> Bob: Another authentication Request
Alice <-- Bob: Another authentication Response
@enduml",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 1,
      "line": 11,
      "offset": 204,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`remark plantuml plugin ignores codeblocks that are not plantuml with lang='' 1`] = `
Object {
  "children": Array [
    Object {
      "lang": null,
      "position": Position {
        "end": Object {
          "column": 4,
          "line": 3,
          "offset": 15,
        },
        "indent": Array [
          1,
          1,
        ],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "code",
      "value": "// Fake",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 4,
      "line": 3,
      "offset": 15,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`remark plantuml plugin ignores codeblocks that are not plantuml with lang='js' 1`] = `
Object {
  "children": Array [
    Object {
      "lang": "js",
      "position": Position {
        "end": Object {
          "column": 4,
          "line": 3,
          "offset": 17,
        },
        "indent": Array [
          1,
          1,
        ],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "code",
      "value": "// Fake",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 4,
      "line": 3,
      "offset": 17,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;
